DuckDuckGo.Fluent.Plugin/HttpCalls.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json;
+using System.Threading.Tasks;
+using static DuckDuckGo.Fluent.Plugin.JsonResult;
+
+namespace DuckDuckGo.Fluent.Plugin
+{
+    public static class HttpCalls
+    {
+        private const string UserAgentString = "Mozilla/5.0 (Windows NT 10.0; Win64; x64)";
+        private static readonly JsonSerializerOptions SerializerOptions = new() { PropertyNameCaseInsensitive = true };
+
+        public static async Task<DuckDuckGoApiResult> GetApiResult(string url)
+        {
+            using var httpClient = new HttpClient();
+            httpClient.Timeout = TimeSpan.FromSeconds(20);
+            httpClient.DefaultRequestHeaders.UserAgent.TryParseAdd(UserAgentString);
+            return await httpClient.GetFromJsonAsync<DuckDuckGoApiResult>(url, SerializerOptions);
```

Comment:
When you have an `async` method the compiler create a state machine that tracks the `async` state (in this case the HTTP call)
Here it's not necessary since you have only one `await` statement.
So you can remove the `async` from the method signature, and the `await` from `httpClient.GetFromJsonAsync` and just return the `Task` object and the user of this method will `await` the returned `Task`.

Reply:
Doing this will break the plugin completely and results in `TaskCanceledException`. Either I've to remove the `using` keyword before `httpClient` or I've to await it.